### PR TITLE
implements list publications api

### PR DESCRIPTION
**Pull request by brunocalza**
Implements the List Publications RPC call.

It goes together with https://github.com/tablelandnetwork/basin-cli/pull/8

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ capnp-rpc = "0.18.0"
 capnpc = "0.18.0"
 chrono = "0.4.31"
 clap = { version = "4.4.4", features = ["derive", "env"] }
-ethers = { git = "https://github.com/avichalp/ethers-rs.git", branch = "avichalp/calibrationnet" }
```
Comment (brunocalza) [inline Cargo.toml:26]: Looks like the branch `avichalp/calibrationnet` no longer exists, and the change was merged. We can safely use original repo

File: lib/protocol/schema/provider.capnp [hunk 2]
```diff
@@ -9,4 +9,6 @@ interface Publications {
         write @0 (chunk :Data);
         done @1 (sig :Data);
     }
+
+    list @3 (owner :Data) -> (publications :List(Text));
```
Comment (brunocalza) [inline lib/protocol/schema/provider.capnp:13]: new interface method

File: lib/evm/src/client.rs [hunk 3]
```diff
@@ -46,4 +46,12 @@ impl EVMClient for BasinClient {
             .map_err(|e| Error::Evm(e.to_string()))?;
         Ok(())
     }
+
+    async fn list_pub(&self, owner: Address) -> Result<Vec<String>, Error> {
```
Comment (brunocalza) [inline lib/evm/src/client.rs:50]: Implementation of the [method of the trait](https://github.com/tablelandnetwork/basin-provider/pull/9/files#diff-754f0b45540b51df93364c3267dd745bc15db3352ddc429b4b8183b5e1d3b351R22)

File: lib/worker/src/rpc.rs [hunk 4]
```diff
@@ -167,6 +167,31 @@ impl<E: EVMClient + 'static> publications::Server for Publications<E> {
             Ok(())
         })
     }
+
+    fn list(
```
Comment (brunocalza) [inline lib/worker/src/rpc.rs:171]: implementation of the RPC list method

File: lib/worker/src/rpc.rs [hunk 5]
```diff
@@ -167,6 +167,31 @@ impl<E: EVMClient + 'static> publications::Server for Publications<E> {
...
+        Promise::from_future(async move {
+            let publications = e.list_pub(owner).await?;
+
+            let mut publication_list = results.get().init_publications(publications.len() as u32);
+            for (i, p) in publications.iter().enumerate() {
+                let p_str: &str = p;
+                publication_list.set(i as u32, p_str.into());
+            }
+
+            Ok(())
```
Comment (avichalp) [inline lib/worker/src/rpc.rs:192]: is it required return `Ok(())` from a async closure?

File: lib/worker/src/rpc.rs [hunk 6]
```diff
@@ -249,7 +274,7 @@ impl publications::callback::Server for UploadCallback {
 }
 
 /// Listens for RPC messages from Basin clients
-pub async fn listen<E: EVMClient>(
+pub async fn listen<E: EVMClient + 'static>(
```
Comment (avichalp) [inline lib/worker/src/rpc.rs:277]: why do we have to change this lifetime annotation?

**Review by avichalp — APPROVED** [review]
looks good!

File: lib/worker/src/rpc.rs (hunk 6 above)
Comment (brunocalza) [reply lib/worker/src/rpc.rs:277]: not really sure. looks like the rustc from CI was on a newer version and complained about this

**Pull request merged**